### ISSUE TITLE
Cart rule not getting triggered properly

### ISSUE THREAD

**Issue by bryandijksterhuis** [cardgate/magento2#80]
Hey y'all,

We're running into an issue with cart rules and the cardgate module after an update a long while back and that problem still persists unfortunately. 

We have a shop for a client of ours that uses the cardgate magento 2 module. The shop is running on M2.4.2-p1 using OneStepCheckout 1.2.051 (the iDev one) and Cardgate Magento2 version 2.0.39. That was the latest version when we started developing and a while ago we figured we'd update to the latest version of Cardgate (2.0.43 as of right now), however that broke something fairly simple. 

We have a cart rule active in Magento that says if this customer is a member of this particular group and is using the ideal cardgate payment method they should get a discount automatically activated of 1.5%. That works absolutely fine on version 2.0.39 of the Cardgate M2 module but it completely breaks on anything above that. We've tried numerous times and one of 2 things will happen: 
1. The customer will get the discount automatically applied when they go to the checkout page (because ideal is the default payment method), but whenever they select a different payment method the discount is not removed. Or....
2. The customer won't get the discount applied when going to the checkout and no matter what payment method you select or what you change, the discount does not get applied. 

Is there any chance you might be able to reproduce this or look into the specific issue? We can reproduce breaking it and fixing it by upgrading or downgrading the cardgate package from 2.0.39 to >= 2.040 so we're fairly certain something broke during one of the updates, but what exactly we're not sure about. 

Hope to hear from you soon,

Kind regards,
Bryan 
on behalf of HC Designs

**Comment by cardgate** [comment]
Hello Brian,

I have tested this issue in the following demo store configuration:

Magento 2.4.3-p1
CardGate Magento2 v2.0.43
CardGate API client library v 1.1.16
 
In this default setup, the checkout order summary is updated as expected, 
and shows the correct discounts upon selecting the different payment methods.

While we only guarantee that our module works in the default setup, we realize 
that OneStepCheckout is a popular module, so I will investigate further regarding 
compatibility with this module.

Kind regards,

Richard Schoots
CardGate

**Comment by bryandijksterhuis** [comment]
Hey Richard,

Have you any news on this ticket? Our client is once more asking about this so any news would be greatly appreciated.

Kind regards,
Bryan

**Comment by cardgate** [comment]
Hello Bryan,

I have tested discounts with the following setup:
Magento 2.4.4
OneStepCheckout 1.2.057

Using only default Magento payment methods, I found that a discount
dependent on a payment method and a customer group is only applied 
when you select the next payment method, and not the payment method 
you initially selected, as it is meant to be.

I have reported this, and since 01-09-2022, version 1.2.058 of 
OneStepCheckout has been released, fixing several payment issues. 
I have tested that release, and I see that this issue has not been fixed yet.

There is a setting in OneStepCheckout, called:
Update totals on payment selection
If this is not selected, the discount value (shown or not) remains static.

It does however not solve the problem I mentioned. In the default checkout, 
our payment methods work as expected in a simulation of your discount 
setup.

Kind Regards

Richard
